neuraxle/steps/caching.py

```diff
@@ -49,29 +50,17 @@ def __init__(
 
         self.cache_folder = cache_folder
 
-    def setup(self, step_path: str, setup_arguments: dict = None):
-        """
-        Fit transform data container using value caching.
-
-        :param setup_arguments: optional additional setup arguments
-        :type setup_arguments: dict
-
-        :param step_path: path of the step in the pipeline ex: `̀pipeline/step_name/`̀
-        :type step_path: str
-
-        :return: tuple(fitted pipeline, data_container)
-        """
-        self.create_checkpoint_path(step_path)
-
-    def handle_fit_transform(self, data_container: DataContainer) -> ('BaseStep', DataContainer):
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> ('BaseStep', DataContainer):
         """
         Fit transform data container.
 
+        :param context: execution context
         :param data_container: the data container to transform
         :type data_container: DataContainer
 
         :return: tuple(fitted pipeline, data_container)
         """
+        self.create_checkpoint_path(context.get_path())
```

Comment:
I think we should remove this as the .save method already creates paths. The checkpoints should also be responsible themselves of creating the paths, or even the whole path tree if it doesn't exist and that there are many nested steps. I don't see why the creation of paths is done here otherwise.

Reply:
the .save method might not have been called yet. resolving